app/src/main/java/edu/grinnell/kdic/MainActivity.java

```diff
@@ -481,5 +525,6 @@ protected void onDestroy() {
         mFavoritesFragment = null;
         mRadioService = null;
         mBoundToRadioService = false;
+        super.onDestroy();
```

Comment:
Also: Why are you nullifying only `mFavoritesFragment` and not the others? You should do it for the others!